src/commands/auth-import.ts

```diff
@@ -0,0 +1,138 @@
+const validateOptions = accountImporter.validateOptions;
+const validateUserJson = accountImporter.validateUserJson;
+const transArrayToUser = accountImporter.transArrayToUser;
+const serialImportUsers = accountImporter.serialImportUsers;
+
+module.exports = new Command("auth:import [dataFile]")
+  .description("import users into your Firebase project from a data file(.csv or .json)")
+  .option(
+    "--hash-algo <hashAlgo>",
+    "specify the hash algorithm used in password for these accounts"
+  )
+  .option("--hash-key <hashKey>", "specify the key used in hash algorithm")
+  .option(
+    "--salt-separator <saltSeparator>",
+    "specify the salt separator which will be appended to salt when verifying password. only used by SCRYPT now."
+  )
+  .option("--rounds <rounds>", "specify how many rounds for hash calculation.")
+  .option(
+    "--mem-cost <memCost>",
+    "specify the memory cost for firebase scrypt, or cpu/memory cost for standard scrypt"
+  )
+  .option("--parallelization <parallelization>", "specify the parallelization for standard scrypt.")
+  .option("--block-size <blockSize>", "specify the block size (normally is 8) for standard scrypt.")
+  .option("--dk-len <dkLen>", "specify derived key length for standard scrypt.")
+  .option(
+    "--hash-input-order <hashInputOrder>",
+    "specify the order of password and salt. Possible values are SALT_FIRST and PASSWORD_FIRST. " +
+      "MD5, SHA1, SHA256, SHA512, HMAC_MD5, HMAC_SHA1, HMAC_SHA256, HMAC_SHA512 support this flag."
+  )
+  .before(requirePermissions, ["firebaseauth.users.create", "firebaseauth.users.update"])
+  .action(async (dataFile, options) => {
+    const projectId = needProjectId(options);
+    const checkRes = validateOptions(options);
+    if (!checkRes.valid) {
+      return checkRes;
+    }
+    const hashOptions = checkRes;
+
+    if (!_.endsWith(dataFile, ".csv") && !_.endsWith(dataFile, ".json")) {
+      return utils.reject("Data file must end with .csv or .json", { exit: 1 });
+    }
+    const stats = await fs.stat(dataFile);
+    const fileSizeInBytes = stats.size;
+    logger.info(`Processing ${clc.bold(dataFile)} (${fileSizeInBytes} bytes)`);
+
+    const batches: any[] = [];
+    let currentBatch: any[] = [];
+    let counter = 0;
+    let userListArr: any[] = [];
+    const inStream = fs.createReadStream(dataFile);
+    if (dataFile.endsWith(".csv")) {
+      userListArr = await new Promise<any[]>((resolve, reject) => {
+        const parser = parse();
+        parser
+          .on("readable", () => {
+            let record: string[] = [];
+            while ((record = parser.read()) !== null) {
+              counter++;
+              const trimmed = record.map((s) => {
+                const str = s.trim().replace(/^["|'](.*)["|']$/, "$1");
+                return str === "" ? undefined : str;
+              });
+              const user = transArrayToUser(trimmed);
+              const err = _.get(user, "error");
+              if (err) {
```

Comment:
We can get rid of the previous line and just make this `if (user.error)`

Reply:
Sure.